src/classes/OPP_StageMappingCMT.cls

```diff
@@ -0,0 +1,193 @@
+
+        // Only create the finalized opp stage to state record if there's no existing finalized record. 
+        Integer finalizedCount = [SELECT Count() FROM Opportunity_Stage_To_State_Mapping__mdt Where Opportunity_State__c =: FINALIZED_STATE_VAL];
+        
+        // Take the value, if there is one, and create a finalized Opp Stage to State custom metadata type record. 
+        if (finalizedCount == 0 && 
+            (contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != null && 
+            contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != '')) {
+            
+            metadataToInsert.add(createCustomMetadata(contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c, FINALIZED_STATE_VAL));
+        }
+
+        return metadataToInsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates the initial Opportunity Stage to State record if there are no existing records. 
+    * When there are no existing records, it will create a new "Committed" state record for a "Pledged" stage
+    * if there is one. 
+    ********************************************************************************************************/
+    public static List<Metadata.CustomMetadata> createPledgedOppStageToStateRecord(List<Metadata.CustomMetadata> metadataToInsert) {
+        npe01__Contacts_And_Orgs_Settings__c contactAndOrgCustomSetting = UTIL_CustomSEttingsFacade.getOrgContactsSettings();
+
+        
+        Integer overallCount = [SELECT Count() FROM Opportunity_Stage_To_State_Mapping__mdt];
+
+        // Get all opportunity stages to see if there is a pledged stage
+        Schema.DescribeFieldResult fieldResult = Opportunity.StageName.getDescribe();
+        List<Schema.PicklistEntry> picklistEntryList = fieldResult.getPicklistValues();
+        Boolean hasPledgedValue = false;
+        String normalPledgeValue = '';
+
+        for (Schema.PicklistEntry picklistEntry : picklistEntryList) {
+            if (picklistEntry.getValue().toUpperCase() == PLEDGED_STAGE) {
+                normalPledgeValue = picklistEntry.getValue();
+                hasPledgedValue = true;
+                break;
+            }
+        }       
+        
+        // Take the value, if there is one, and create a finalized Opp Stage to State custom metadata type record. 
+        // Only create the pledged opp stage to state record if there's no existing opportunity stage to state records at all. 
+        if (overallCount == 0 && hasPledgedValue) {
+            metadataToInsert.add(createCustomMetadata(normalPledgeValue, COMMITTED_STATE_VAL));
+        }
+
+        return metadataToInsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a custom metadata record based on the parameters provided to the method. 
```

Comment:
Please add the @param entries for the parameters and the @return entry for the method.